### DISCUSSION TITLE
[Github Projects] Setting a default value for a project custom field

### DISCUSSION THREAD

**Discussion by djkeh** [community/community#72181] [Projects and Issues]
### Select Topic Area

Product Feedback

### Feature Area

Projects

### Body

Currently it is not possible to set a default value for each custom field, so you have to set all the fields up every time you create a project issue. This could be done even more convenient with the default value option. We can currently do it for the `status` field only at the project workflows, however, the workflows provide very limited options.

## Reference

* https://github.com/orgs/community/discussions/8076

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by nfuzer** [comment]
+1 to this 

If we can set default fields when new issues are created, this would help tremendously.  e.g., the custom field  "Squad" = Squad 1 for a given project.  Multiple squads sometimes use the same repo, making project management harder.  This would be a welcomed enhancement.

**Comment by luetm** [comment]
A shame that user feedback is being ignored :(

**Reply by SergiLIFE** [reply]
I am a paying GitHub customer and Microsoft Azure ISV (L.I.F.ECoach121.com Limited, repo L-I-F-ECoach121-com-Limited/SergiLIFE-life-azure-system) whose development and CI/CD stack is built around GitHub, GitHub Actions, and GitHub Copilot. For more than 8 weeks, unresolved Copilot API failures, repeated account restrictions, and blocking “corruption pattern” checks in my main repo have prevented me from progressing a production‑grade Azure Marketplace ISV offer that is already in active partnership discussions with Microsoft, Google, IBM, and Kortext. This directly jeopardises my Q4 2025/Q1 2026 launch, my pre‑revenue/seed funding round, and multiple cloud‑provider relationships.

**Comment by redcatbear** [comment]
GitHub custom fields are a very useful addition. It is strange that a (seemingly) trivial feature like a default for field values is not present.

We have a situation where most of the fields could be at the default value 80% of the time, so having the defaults would be a big time saver and improve the user experience. And, I venture a guess that this situation is quite common.

Imagine fields like `urgency` and `criticality`. In a good project those would be at `low` except for a few special cases.

**Reply by rfc2822** [reply]
Have exactly that use-case, an "urgency" field that should be "standard" by default.

**Comment by azenMatt** [comment]
👋 

Would this change be primarily useful for single-select fields, or is this something that would be valuable across all field types? If it would be valuable for text number or date fields, what are the use cases you would have a default for?

**Reply by rfc2822** [reply]
I think it would provide more flexibility for Projects users if a default value would be possible for all fields. Most useful use-cases for me:

- single-select: for instance a priority field that should be "low" or "standard" by default
- number: same, when the priority is a number
- text: there could a _Notes_ field, which could be used to set it to a value like "auto-added" when it's set by the Auto-add workflow. Could be interesting to see how it was added quickly.
- iteration: could be useful when adding issues into Project (= into backlog) as start of current/next iteration → default values "current iteration", "next iteration", "iteration after next iteration" could be useful

**Reply by redcatbear** [reply]
Everything that has a primitive value should definitely get an optional default value:
* Single selects
* Numbers
* Strings

Multiple selects would be nice, but are not as important as the other ones for the beginning. I'd rather have the MVP soon than needing to wait for the gold plated variant longer. :-)

**Reply by djkeh** [reply]
All field types, of course! Every custom value should be able to be set by user. Examples be like:

* text: ` `
* number: `0`
* date: today
* single-select: one of the options
* iteration: current iteration

**Reply by Or1g3n** [reply]
Please implement this feature!
